src/Test.UnitTests.Sarif/Core/AddressTests.cs

```diff
@@ -0,0 +1,201 @@
+
+            address.AbsoluteAddress = long.MinValue;
+            Address_VerifyRoundTripFromObjectHelper(address,
+                shouldSerialize: false, reconstructedAbsoluteAddress: -1);
+            address.AbsoluteAddress--;
+            Address_VerifyRoundTripFromObjectHelper(address,
+                shouldSerialize: false, reconstructedAbsoluteAddress: -1);
+
+            address.AbsoluteAddress = int.MinValue;
+            Address_VerifyRoundTripFromObjectHelper(address,
+                shouldSerialize: false, reconstructedAbsoluteAddress: -1);
+            address.AbsoluteAddress--;
+            Address_VerifyRoundTripFromObjectHelper(address,
+                shouldSerialize: false, reconstructedAbsoluteAddress: -1);
+
+            address.AbsoluteAddress = -2;
+            Address_VerifyRoundTripFromObjectHelper(address,
+                shouldSerialize: false, reconstructedAbsoluteAddress: -1);
+
+            address.AbsoluteAddress = -1;
+            Address_VerifyRoundTripFromObjectHelper(address,
+                shouldSerialize: false, reconstructedAbsoluteAddress: -1);
+
+            address.AbsoluteAddress = 0;
+            Address_VerifyRoundTripFromObjectHelper(address,
+                shouldSerialize: true, reconstructedAbsoluteAddress: address.AbsoluteAddress);
+
+            address.AbsoluteAddress = 1;
+            Address_VerifyRoundTripFromObjectHelper(address,
+                shouldSerialize: true, reconstructedAbsoluteAddress: address.AbsoluteAddress);
+
+            address.AbsoluteAddress = 2;
+            Address_VerifyRoundTripFromObjectHelper(address,
+                shouldSerialize: true, reconstructedAbsoluteAddress: address.AbsoluteAddress);
+
+            address.AbsoluteAddress = int.MaxValue;
+            Address_VerifyRoundTripFromObjectHelper(address,
+                shouldSerialize: true, reconstructedAbsoluteAddress: address.AbsoluteAddress);
+            address.AbsoluteAddress++;
+            Address_VerifyRoundTripFromObjectHelper(address,
+                shouldSerialize: true, reconstructedAbsoluteAddress: address.AbsoluteAddress);
+
+            address.AbsoluteAddress = long.MaxValue;
+            Address_VerifyRoundTripFromObjectHelper(address,
+                shouldSerialize: true, reconstructedAbsoluteAddress: address.AbsoluteAddress);
+            address.AbsoluteAddress++;
+            Address_VerifyRoundTripFromObjectHelper(address,
+                shouldSerialize: true, reconstructedAbsoluteAddress: address.AbsoluteAddress);
+
+            address.AbsoluteAddress = ulong.MaxValue;
+            Address_VerifyRoundTripFromObjectHelper(address,
+                shouldSerialize: true, reconstructedAbsoluteAddress: address.AbsoluteAddress);
+            address.AbsoluteAddress++;
+            Address_VerifyRoundTripFromObjectHelper(address,
+                shouldSerialize: true, reconstructedAbsoluteAddress: address.AbsoluteAddress);
```

Comment:
This looks like a list of test cases all being put through the same helper method. Can we consolidate this test a little so it conforms to "Arrange/Act/Assert" a little more cleanly?

Reply:
looking

Reply:
checked and made the code clearer.

when I have the pattern in mind it make me realize that a few var name is not clear e.g. a AbsoluteAddress that is the expected result to compare to, after the "Act", I have changed the name to "expectedAbsoluteAddress" to make it more clear.